solutions/usaco-1206.mdx

```diff
@@ -0,0 +1,107 @@
+---
+id: usaco-1206
+source: USACO Silver 2022 February
+title: Redistributing Gifts
+author: Chongtian Ma
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob1_silver_feb22.html)
+
+## Explanation
+
+Similar to the official analysis, we created a directed graph with edges from $i$ to $j$ if cow $i$ prefers gift $j$ more compared to his own. 
+If cow $i$ is able to receive a gift where it has an outgoing edge, then the solution will always be more optimal. 
+Remember to also add an edge between cow $i$ and $i$ in the case where it cannot get a more preferred gift.
+
+Then, I separated the graph into strongly connected components. If $i$ is in a strongly connected component, this means that cow $i$ can receive
+all gifts in the strongly connected component through a series of reassignments and is guaranteed to have one or more preferred gift (including itself) based 
+on SCC definition. Obviously you only want to assign cow $i$ with a more preferred gift.
+## Implementation
+
+I used Kosaraju's algorithm to find all strongly connected components in the graph. At the end, I just went down cow $i$'s preference list and assigned the first gift belonging in the SCC.
```

Comment:
don't use first person

Reply:
not addressed

Reply:
fixed @SansPapyrus683 
By the way is it also okay if I use "we"?